cmux.go

```diff
@@ -103,22 +107,21 @@ func (m *cMux) Serve() error {
 			continue
 		}
 
-		go m.serve(c)
+		wg.Add(1)
+		go m.serve(c, &wg)
 	}
 }
 
-func (m *cMux) serve(c net.Conn) {
+func (m *cMux) serve(c net.Conn, wg *sync.WaitGroup) {
+	defer wg.Done()
+
 	muc := newMuxConn(c)
 	for _, sl := range m.sls {
 		for _, s := range sl.ss {
 			matched := s(muc.sniffer())
 			muc.reset()
 			if matched {
-				select {
-				case sl.l.connc <- muc:
-				case <-sl.l.donec:
-					_ = c.Close()
-				}
+				sl.l.connc <- muc
```

Comment:
> what if serve is called for more than len(sl.l.connc) and the go-routine that actually calls sl.l.Accept() is dead.

Doesn't seem like a problem to me; the connection would still be sent on `sl.l.connc` which is buffered to 1024. What am I missing?